Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+                    // Sets the currentFloor to the new floor level and adds one, since location uses
+                    // zero based flooring system.
+                    currentFloor = floorLevel + 1
+                    // The floor manager is used to filter so that only the data from the current floor is
+                    // displayed to the user.
+                    if let floorManager = map.floorManager {
+                        floorManager.levels.forEach {
+                            $0.isVisible = currentFloor == $0.levelNumber
+                        }
+                    }
+                }
+                // This indicates whether the location data was sourced from GNSS (Satellites), BLE (Bluetooth Low Energy)
+                // or AppleIPS (Apple's proprietary location system.
+                let source = location.additionalSourceProperties[.positionSource] as? String ?? ""
+                switch source {
+                case "GNSS":
+                    satelliteCount = location.additionalSourceProperties[.satelliteCount] as? Int ?? 0
+                default:
+                    sensorCount = location.additionalSourceProperties[.transmitterCount] as? Int ?? 0
+                }
+                horizontalAccuracy = location.horizontalAccuracy
+                // Updates every time the location changes.
+                getStatusLabelText(source: source)
+            }
+        }
+        
+        /// Updates the labels on the view with the current state of the indoors data source.
+        private func getStatusLabelText(source: String) {
+            labelTextLeading = ""
+            labelTextTrailing = ""
+            if let currentFloor {
+                labelTextLeading += "Current floor: \(currentFloor)\n"
+                if let horizontalAccuracy {
+                    var horizontalAccuracyMeasurement = Measurement<UnitLength>(value: horizontalAccuracy, unit: .meters)
+                    labelTextLeading += "Accuracy: \(horizontalAccuracyMeasurement.formatted(.distance))"
+                }
+                if let sensorCount {
+                    labelTextTrailing += "Number of sensors: \(sensorCount)\n"
+                } else if let satelliteCount, source == "GNSS" {
+                    labelTextTrailing += "Number of satellites: \(satelliteCount)\n"
+                }
+                labelTextTrailing += "Data source: \(source)"
+            } else {
+                labelTextLeading = "No floor data."
+            }
+        }
+    }
+}
+
+private extension URL {
+    static var indoorsMap: URL {
+        URL(string: "https://www.arcgis.com/home/item.html?id=8fa941613b4b4b2b8a34ad4cdc3e4bba")!
+    }
+}
+
+private extension FormatStyle where Self == Measurement<UnitLength>.FormatStyle {
+    /// The format style for the distances.
+    static var distance: Self {
+        .measurement(width: .abbreviated, usage: .asProvided, numberFormatStyle: .number.precision(.fractionLength(2)))
+    }
+}
```

Comment:
Data formatting logic in the model is a big code smell. Data should be formatted based on the current locale, but the model doesn't have that information (because it comes from the environment of the view).